src/main/java/org/packetfence/agent/MainActivity.java

```diff
@@ -494,6 +495,14 @@ public void configureWirelessConnectionWPAPEAPAndEAPTLS(Object[] categoryObj, Ha
                         MainActivity.this.userP12Name = (String) config.get("PayloadDisplayName");
                         MainActivity.this.tlsUsername = (String) config.get("PayloadCertificateFileName");
                     }
+                    if (payloadType.equals("com.apple.security.pkcs1")) {
+                        showInDebug("Found the EAP-TLS root certificate");
+                        MainActivity.this.serverCN = (String) config.get("PayloadCertificateFileName");
+                    } else {
+                        if (MainActivity.this.api_version >= 29) {
+                            showInBox("Your PacketFence server needs a patch or change manually the html/captive-portal/templates/wireless-profile-tls.xml and allow access to 'com.apple.security.pkcs1'");
```

Comment:
I think telling that to end-users is 'useless' - better just say "Your Android version..... blabla. Please contact your system administrator."

Reply:
OK